src/evaluator/program.cpp

```diff
@@ -502,6 +593,69 @@ void Call::display(std::ostream& stream, size_t depth) const {
 
 bool Call::can_evaluate_to_function() const { return true; }
 
+Cond::Cond(
+    std::unique_ptr<Expression> condition,
+    std::unique_ptr<Expression> then,
+    std::unique_ptr<Expression> otherwise)
+    : Expression(), condition(std::move(condition)), then(std::move(then)), otherwise(std::move(otherwise)) {}
+
+std::unique_ptr<Cond> Cond::parse(std::shared_ptr<ast::List> arguments) {
+    auto cons = arguments->to_cons();
+
+    if (!cons) {
+        throw std::runtime_error("`cond` takes 2 or 3 arguments, provided 0");
+    }
+
+    auto condition = Expression::from_element(cons->left);
+
+    cons = cons->right->to_cons();
+
+    if (!cons) {
+        throw std::runtime_error("`cond` takes 2 or 3 arguments, provided 1");
+    }
+
+    auto then = Expression::from_element(cons->left);
+
+    cons = cons->right->to_cons();
+
+    std::unique_ptr<Expression> otherwise;
+    if (!cons) {
+        otherwise = Expression::from_element(std::make_shared<Null>(Null(arguments->span)));
+    }
+    else {
+        otherwise = Expression::from_element(cons->left);
+    }
```

Comment:
```suggestion
    if (cons) {
        otherwise = Expression::from_element(cons->left);
    } else {
        otherwise = Expression::from_element(std::make_shared<Null>(Null(arguments->span)));
    }
```